app/src/main/java/com/example/reduxtestapp/ui/country/CountryScreen.kt

```diff
@@ -0,0 +1,245 @@
+package com.example.reduxtestapp.ui.country
+
+import androidx.compose.foundation.layout.Arrangement
+import androidx.compose.foundation.layout.Column
+import androidx.compose.foundation.layout.fillMaxSize
+import androidx.compose.foundation.layout.fillMaxWidth
+import androidx.compose.foundation.layout.padding
+import androidx.compose.foundation.lazy.LazyColumn
+import androidx.compose.foundation.lazy.items
+import androidx.compose.foundation.text.KeyboardOptions
+import androidx.compose.material.icons.Icons
+import androidx.compose.material.icons.filled.Search
+import androidx.compose.material3.Button
+import androidx.compose.material3.CircularProgressIndicator
+import androidx.compose.material3.Icon
+import androidx.compose.material3.Scaffold
+import androidx.compose.material3.Text
+import androidx.compose.material3.TextField
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.LaunchedEffect
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.runtime.remember
+import androidx.compose.runtime.setValue
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.res.stringResource
+import androidx.compose.ui.text.input.ImeAction
+import androidx.compose.ui.text.style.TextAlign
+import androidx.compose.ui.tooling.preview.Preview
+import androidx.compose.ui.unit.dp
+import com.example.reduxtestapp.R
+import com.example.reduxtestapp.redux.Action
+import com.example.reduxtestapp.redux.AppState
+import com.example.reduxtestapp.redux.CountryState
+import com.example.reduxtestapp.ui.theme.ReduxTestAppTheme
+import com.ramcosta.composedestinations.annotation.Destination
+import com.ramcosta.composedestinations.annotation.RootNavGraph
+import kotlinx.coroutines.delay
+import org.koin.compose.koinInject
+import org.reduxkotlin.Store
+
+@RootNavGraph
+@Destination
+@Composable
+fun CountryScreen(
+    modifier: Modifier = Modifier,
+    store: Store<AppState> = koinInject(),
+) {
+    var uiState by remember {
+        mutableStateOf(CountryViewState(
+            status = CountryState.Status.PENDING
+        ))
+    }
+
+    store.subscribe{
+        uiState = store.state.toCountryViewState()
+    }
```

Comment:
this pattern will repeat for every screen, so some util for subscribing to state would be useful

so that you can do something like this:

```
val uiState = store.collectState().toCountryViewState()
```

bonus points if this mapping between AppState and ViewState can be hidden away behind `collectState`. We can discuss this